### PR TITLE
Feature/multi queries

### PR DESCRIPTION
**Pull request by marianogaragiola**

### CODE REVIEW

File: splight_lib/models/datalake.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+
+class TraceType(str, Enum):
+    QUERY = "QUERY"
+    EXPRESSION = "EXPRESSION"
+    METADATA = "METADATA"
+
+    @classmethod
+    def choices(cls):
+        return tuple((i.name, i.value) for i in cls)
```
Comment (francozacco) [inline splight_lib/models/datalake.py:43]: I think this is already defined on `splight_lib/client/datalake/constants.py`

File: splight_lib/models/datalake.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    to_timestamp: datetime | None = None
+    traces: list[Trace] = []
+
+    _dl_client: AbstractDatalakeClient = PrivateAttr()
+    _traces_ref: dict[str, dict] = {}
+
+    def add_trace(self, trace: Trace) -> None:
+        self.traces.append(trace)
+        if trace.ref_id in self._traces_ref:
+            raise TraceAlreadyExistsError(trace.ref_id)
```
Comment (francozacco) [inline splight_lib/models/datalake.py:109]:
```suggestion
        if trace.ref_id in self._traces_ref:
            raise TraceAlreadyExistsError(trace.ref_id)
        self.traces.append(trace)
```

File: splight_lib/models/datalake_base.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        cls,
+        asset: str | Asset,
+        attribute: str | Attribute,
+        extra_pipeline: list[dict[str, Any]] = {},
+        **params: Dict,
+    ) -> list[Self]:
+        request = _to_data_request(
+            cls, asset, attribute, extra_pipeline, **params
+        )
+        request.add_trace(Trace.from_address(asset, attribute))
```
Comment (francozacco) [inline splight_lib/models/datalake_base.py:37]: Why do we add the same Trace again?

File: splight_lib/models/datalake_base.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        trace.add_step(PipelineStep.from_dict(step))
+    request.add_trace(trace)
+    return request
+
+
+def datalake_model_serializer(data: Dict, default=str, **dumps_kwargs):
+    new_data = {
+        k: v if not isinstance(v, dict) else v["id"] for k, v in data.items()
+    }
+    return json.dumps(new_data, default=default, **dumps_kwargs)
```
Comment (francozacco) [inline splight_lib/models/datalake_base.py:126]: Where is this used?

File: splight_lib/models/datalake.py (hunk 1 above)
Comment (marianogaragiola) [reply splight_lib/models/datalake.py:43]: Ups, forgot to remove it from constants.

File: splight_lib/models/datalake_base.py (hunk 3 above)
Comment (marianogaragiola) [reply splight_lib/models/datalake_base.py]: My bad, it shouldn't

File: splight_lib/models/datalake_base.py (hunk 4 above)
Comment (marianogaragiola) [reply splight_lib/models/datalake_base.py]: Is not being used anymore, I can remove it

**Review by francozacco — APPROVED** [review]

**Review by smartinellimarco — APPROVED** [review]

**Pull request merged**